### ISSUE TITLE
Accuracy is low

### ISSUE THREAD

**Issue by Benybrahim** [serengil/deepface#260]
Hello,

First, Thank you for the efforts that you are making for this repo. It is really great work.

So, my question is as follow: I'm trying to reproduce the result for LFW dataset to get around 99% accuracy. but I get only 75-77% accuracy.

I used different model including `FaceNet` and `Dlib`, and also different detector like `Retinaface` and `opencv`, but still I got almost same results.

Thank you for in advance for your insights.

Here is my code:

```python
from deepface import DeepFace
from sklearn.datasets import fetch_lfw_pairs
from sklearn.metrics import classification_report


### DATA
data = fetch_lfw_pairs(subset = 'test', color = True, resize = 1)
pairs = data.pairs
labels = data.target
imgs = []
for img1, img2 in pairs:
    img1 = img1[:, :, ::-1].astype("uint8") # rgb to rbg
    img2 = img2[:, :, ::-1].astype("uint8")
    imgs.append([img1, img2])


### MODEL
params = {
    "model_name": "Facenet",
    "detector_backend": "retinaface",
    "distance_metric": "euclidean",
#   "enforce_detection": False
}
results = DeepFace.verify(imgs, **params)

### EVALUATION
actuals = [True if label == 1 else False for label in labels]
predictions = [res["verified"] for res in results.values()]
print(classification_report(labels, predictions))
```

**Note**: I also used the original LFW dataset since the images are larger, but it still give the same result.

**Comment by serengil** [comment]
I mentioned this in one of my blog post and vlog as well.

Notebook: https://github.com/serengil/tensorflow-101/blob/master/python/LFW.ipynb (Dlib got 92.57% here)

Blog post: https://sefiks.com/2020/08/27/labeled-faces-in-the-wild-for-face-recognition/ (same model and accuracy with notebook)

Vlog: https://youtu.be/9fT-9TFjeDQ (ArcFace got 93% here)

BTW, deepface using pre-trained models. In other words, I did not train them by myself. I think the reference studies do subsampling in LFW data set. Besides, they are using their own detection and alignment methods. This causes the difference.

**Comment by Benybrahim** [comment]
I know that that deepface only use pre-trained models, but I was wondering why I couldn't reproduce the 92.75% using Dlib following your tutorial that I also saw and followed before asking the questio.

If I get 92% is enough, but 75% mean that something is wrong with the versioning or something else. Especially, that I got 98% accuracy using other framework on the same dataset. But, I do prefer deep face. That is why I insisted on making it work and ask this question.

**Comment by serengil** [comment]
This is very interesting. Did you apply the same procedures described in this video?

https://youtu.be/9fT-9TFjeDQ

**Comment by serengil** [comment]
That line attracts my attention -> astype("uint8")

That might be reason of your trouble. I didn't do it in my notebook: https://github.com/serengil/tensorflow-101/blob/master/python/LFW.ipynb

**Comment by serengil** [comment]
Please apply same procedures in the video because I call verify function in a for loop. 

Results might store items unsorted. E.g. 1 then 10 instead of 1 then 2

**Comment by bml1g12** [comment]
Thank you for your quick responses here - really appreciate it! Looks like a fantastic repo. But I am also running into some trouble. 

I tried running[ the notebook](https://github.com/serengil/tensorflow-101/blob/master/python/LFW.ipynb) as is (with no changes except `!pip install dlib, deepface` to get the non-standard libraries) and got 
```
ValueError: Face could not be detected. Please confirm that the picture is a face photo or consider to set enforce_detection param to False.
```

If I set this to False I get
`TypeError: Cannot handle this data type: (1, 1, 3), <f4`

Are you able to advise what version of the libraries the notebook was running at previously? I notice in the comments of the article you linked someone has had the same issue as me: https://sefiks.com/2020/08/27/labeled-faces-in-the-wild-for-face-recognition/
